### PR TITLE
Adding 8 new metafield icons

### PR DESCRIPTION
**Pull request by leileu**
### WHY are these changes introduced?

We are updating the type picker for Metafields to have unique iconography for each type.
These addition of icons will relate to fixing this issue [#495](https://github.com/Shopify/metafield-issues/issues/495)

These icons are also added into the [Polaris Icon Library](https://www.figma.com/file/mMHFt3kEDNjLMZWowi6gnt/Polaris-Icons?node-id=11100%3A286&t=xlthiQQtDUNlMNf6-1) and merged. 

### WHAT is this pull request doing?

Adding 8 new Metafield icons
- Decimal
- Weight
- Volume
- Rich text
- Calendar time
- Page reference
- Product reference
- Collection reference

### CODE REVIEW

**Review by sarahill — APPROVED** [review]

**Comment by leileu** [comment]
@kyledurand Did you get a chance to take a look at this

**Comment by kyledurand** [comment]
Tests are failing because the names are incorrect. Can you add `Minor` suffix to this icons? i.e. `CalendarTime` should be `CalendarTimeMinor`

**Comment by leileu** [comment]
> Tests are failing because the names are incorrect. Can you add `Minor` suffix to this icons? i.e. `CalendarTime` should be `CalendarTimeMinor`

Thanks Kyle!! I just updated them

File: .changeset/blue-ducks-stare.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': minor
+---
+
+Adding 8 minor icons for metafields
```
Comment (kyledurand) [inline .changeset/blue-ducks-stare.md:5]:
```suggestion
Added 8 minor icons for metafields
```

**Review by kyledurand — APPROVED** [review]
Looks good! Just added a small comment around the changelog, we try to use past tense. Other than that 🚢 !

**Pull request merged**